app/src/main/java/com/karumi/data/repository/SuperHeroDataSource.kt

```diff
@@ -0,0 +1,15 @@
+package com.karumi.data.repository
+
+import com.karumi.domain.model.DomainError
+import com.karumi.domain.model.SuperHero
+import org.funktionale.either.Either
+
+interface SuperHeroDataSource {
```

Comment:
This is something I've been running away for ages now. Why do we need an interface for two data sources that are so different? There are methods here that do not make sense for the `NetworkDataSource` (`isUpdated`, `populate` and `contains`). 

The repository is meant to hide data sources from its clients but not from itself so lately I've been having repositories having `XCacheDataSource` and `XRemoteDataSource` as dependencies for `XRepository` but there is no common interface/class between them, they have different uses so I keep them separate (instead of having a list of abstract data sources) .

Reply:
can we tomorrow talk about that because sound good for me and I like write a good solution. I don't like my current solution and ideas are always welcome.